src/celeritas/neutron/interactor/detail/CascadeParticle.hh

```diff
@@ -0,0 +1,39 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/neutron/interactor/detail/CascadeParticle.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Types.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/Types.hh"
+#include "celeritas/phys/FourVector.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Particle data to track the intra-nuclear cascade interactions.
+ */
+struct CascadeParticle
+{
+    //! Types of cascade particles in intra-nuclear interactions
+    enum class ParticleType
+    {
+        unknown,
+        proton,
+        neutron,
+        gamma
+    };
+
+    ParticleType type{ParticleType::unknown};  //!< Particle type
+    units::MevMass mass;  // !< Particle mass
+    FourVector vec4;  //!< Four momentum in natural MevMomentum and
+                      //!< MevEnergy units
```

Comment:
I'm not a big fan of the combined text+number; could it just be "momentum"?

Reply:
Change vec4 to four_vec.